### ISSUE TITLE
Update scripture-resources-rcl in GWE for Greek highlight fix

### ISSUE THREAD

**Issue by elsylambert** [unfoldingWord/gateway-edit#570]
Need to update scripture-resources-rcl version to incorporate the fix for highlighting in GWE.

Message from Stepheh in Zulip: 

> I'd like to revisit the highlighting problems in tCC and GE whenever the Quote field in a TN includes a Greek word with an elision marks. This is causing serious difficulties for our proofreaders, which means it will cause the same difficulties for end users. For example, here's a section of the proofreading spreadsheet:
![image](https://github.com/unfoldingWord/scripture-resources-rcl/assets/4523496/14bd4929-968a-4c6f-b589-7aaa17151d34)
Every single one of these rows is a place where the highlighting is incorrect because tCC and GE are not recognizing the Greek words with elision marks. Here's an example of what it looks like in tCC:
![image (1)](https://github.com/unfoldingWord/scripture-resources-rcl/assets/4523496/0133e4e7-d7c0-4061-bf4b-62d8bfa1feb1)
I've included every Greek word in the Quote field, but every word with an elision mark fails to highlight. How is the program recognizing words in the UGNT based on the Quote field? Is it stripping out punctuation when it looks for a matching string of characters?
Fixing this issue will make the resources much more useable, so I appreciate any help.

**Comment by elsylambert** [comment]
This PR fixes the issue in scripture-resource-rcl https://github.com/unfoldingWord/scripture-resources-rcl/pull/180

**Comment by birchamp** [comment]
#535 will need to be retested when this is complete. @elsylambert

**Comment by abelpz** [comment]
@kintsoogi You may need to update that library in single-scripture-rcl too. @PhotoNomad0 may be able to point you in the right direction.

**Comment by PhotoNomad0** [comment]
@kintsoogi I looks like the additional change you will need to make is in ScriptureCard.  The new SelectionsContextProvider props are:

```
SelectionsContextProvider.propTypes = {
  /** words in a selection */
  selections: PropTypes.instanceOf(Map),
  /** action taken after a selection is made */
  onSelections: PropTypes.func.isRequired,
  /** the quote to be selected */
  quote: PropTypes.string.isRequired,
  /** all chapter-verses where quote may be found */
  bookObject: PropTypes.object,
  /** string chapter-verse reference. i.e. 1:5-6 */
  refString: PropTypes.string,
  /** if quote occurs mulitple times, this is the occurence of the one selected */
  occurrence: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** action taken when quote is provided */
  onQuote: PropTypes.func,
};
```

Currently we are passing:
```
  return (
    <SelectionsContextProvider
      selections={selections}
      onSelections={newSelections => {
        // console.log('onSelections', newSelections)
      }}
      quote={selectedQuote?.quote}
      occurrence={fixOccurrence(selectedQuote?.occurrence)}
      verseObjectsMap={verseObjectsMap}
    >
```

**Comment by PhotoNomad0** [comment]
Part of the challenge is that in the UGNT we are not actually  using an `elision` character, but using the `RIGHT SINGLE QUOTATION MARK` (U+2019) . So where that would normally be treated as punctuation, we have to add special handling to include it as part of the word in the original language.

**Comment by abelpz** [comment]
> Part of the challenge is that in the UGNT we are not actually  using an `elision` character, but using the `RIGHT SINGLE QUOTATION MARK` (U+2019) . So where that would normally be treated as punctuation, we have to add special handling to include it as part of the word in the original language.

This is fixed with the latest version of scripture-resources-rcl and working in tc-create. I think you are right pointing to the change in the API for the selections component @PhotoNomad0, changing that should fix it.

**Comment by PhotoNomad0** [comment]
> > Part of the challenge is that in the UGNT we are not actually  using an `elision` character, but using the `RIGHT SINGLE QUOTATION MARK` (U+2019) . So where that would normally be treated as punctuation, we have to add special handling to include it as part of the word in the original language.
> 
> This is fixed with the latest version of scripture-resources-rcl and working in tc-create. I think you are right pointing to the change in the API for the selections component @PhotoNomad0, adapting to that should fix it. No need to do any special handling in gE or other library.

Made progress with the update to the props for the selections context.  At least now highlighting is working on most tNotes.  But still this specific issue is not working even with the newer rcl.  Double checking to see if I have missed one detail.

OK found the problem is in getQuoteMatchesInBookRef() which is stripping out the right quote character here

https://github.com/unfoldingWord/uw-quote-helpers/blob/main/src/helpers/quote.js#L244

**Comment by PhotoNomad0** [comment]
@elsylambert @danielklapp The fix is in release v2.2.0 build 3179311 QA
